### PR TITLE
graphQl-892: Improved strict typing in CartAddressInterface

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description (*)
Improved strict typing in CartAddressInterface.

### Fixed Issues (if relevant)
1. #892: [Checkout] Improve strict typing in CartAddressInterface

### CODE REVIEW

**Comment by VitaliyBoyko** [comment]
Looks like we have the same issue for the billing address as for the shipping address here #864.

**Comment by VitaliyBoyko** [comment]
Hi @paliarush please clarify what should be returned in the scope of this case \Magento\GraphQl\Quote\Customer\GetSpecifiedBillingAddressTest::testGetSpecifiedBillingAddressIfBillingAddressIsNotSet. So far it returns object with nullable fields. As far as I have understood it should be null instead of an object.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
@VitaliyBoyko billing address should be `null`. Please consider fixing api-functional test.

File: app/code/Magento/QuoteGraphQl/Model/Resolver/BillingAddress.php [hunk 1]
```diff
@@ -44,7 +44,13 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
         $cart = $value['model'];
 
         $billingAddress = $cart->getBillingAddress();
-        if (null === $billingAddress) {
+        if (null === $billingAddress ||
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/BillingAddress.php]: Could you please use `validateAddressFromSchema` method? I mean move it to the separate class from  `\Magento\QuoteGraphQl\Model\Resolver\ShippingAddresses` and reuse for both - shipping and billing.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please see my suggestion with `validateAddressFromSchema` method.

**Review by lenaorobei — APPROVED** [review]

**Pull request merged**